crates/sui-benchmark/src/lib.rs

```diff
@@ -113,12 +113,62 @@ pub struct LocalValidatorAggregatorProxy {
 }
 
 impl LocalValidatorAggregatorProxy {
-    pub fn from_auth_agg(agg: Arc<AuthorityAggregator<NetworkAuthorityClient>>) -> Self {
+    pub async fn from_genesis(
+        genesis: &Genesis,
+        registry: &Registry,
+        fullnode_rpc_url: &str,
+    ) -> Self {
+        let (aggregator, _) = AuthorityAggregatorBuilder::from_genesis(genesis)
+            .with_registry(registry)
+            .build()
+            .unwrap();
+
+        let committee_store = aggregator.clone_committee_store();
+
+        let reconfig_observer = FullNodeReconfigObserver::new(
+            fullnode_rpc_url,
+            committee_store,
+            aggregator.safe_client_metrics_base.clone(),
+            aggregator.metrics.clone(),
+        )
+        .await;
+        let quorum_driver_metrics = Arc::new(QuorumDriverMetrics::new(registry));
         let qd_handler =
-            QuorumDriverHandlerBuilder::new(agg, Arc::new(QuorumDriverMetrics::new_for_tests()))
-                // TODO: replace with a real reconfig observer
-                .with_reconfig_observer(Arc::new(DummyReconfigObserver {}))
+            QuorumDriverHandlerBuilder::new(Arc::new(aggregator), quorum_driver_metrics)
+                .with_reconfig_observer(Arc::new(reconfig_observer))
                 .start();
+
+        let qd = qd_handler.clone_quorum_driver();
+        Self {
+            _qd_handler: qd_handler,
+            qd,
+        }
+    }
+
+    pub async fn from_network_config(
+        configs: &NetworkConfig,
+        registry: &Registry,
+        fullnode_rpc_url: &str,
+    ) -> Self {
+        let (aggregator, _) = AuthorityAggregatorBuilder::from_network_config(configs)
+            .with_registry(registry)
+            .build()
+            .unwrap();
+        let committee_store = aggregator.clone_committee_store();
+
+        let reconfig_observer = FullNodeReconfigObserver::new(
+            fullnode_rpc_url,
+            committee_store,
+            aggregator.safe_client_metrics_base.clone(),
+            aggregator.metrics.clone(),
+        )
+        .await;
+        let quorum_driver_metrics = Arc::new(QuorumDriverMetrics::new(registry));
+        let qd_handler =
+            QuorumDriverHandlerBuilder::new(Arc::new(aggregator), quorum_driver_metrics)
+                .with_reconfig_observer(Arc::new(reconfig_observer))
+                .start();
+
```

Comment:
nit: It's possible to refactor the above two functions to avoid duplicating some code here